source/common/http/conn_manager_utility.cc

```diff
@@ -31,7 +31,7 @@ std::string ConnectionManagerUtility::determineNextProtocol(Network::Connection&
   // See if the data we have so far shows the HTTP/2 prefix. We ignore the case where someone sends
   // us the first few bytes of the HTTP/2 prefix since in all public cases we use SSL/ALPN. For
   // internal cases this should practically never happen.
-  if (-1 != data.search(Http2::CLIENT_MAGIC_PREFIX.c_str(), Http2::CLIENT_MAGIC_PREFIX.size(), 0)) {
+  if (0 == data.search(Http2::CLIENT_MAGIC_PREFIX.c_str(), Http2::CLIENT_MAGIC_PREFIX.size(), 0)) {
```

Comment:
How about using `copyOut` to a stack allocated char[] and compare with `CLIENT_MAGIC_PREFIX`? `CLIENT_MAGIC_PREFIX` is short so that shouldn't cost much.

Reply:
I wonder if we even need to handle arbitrary points, and can just have something like a prefix check.  I suspect that's what we want most of the time.

Either way I'm happy with this change as an incremental improvement.  My one request would be a release note (version history note, whaever) just because it fixes a bug on the data plane, and no one knows if someone is depending on this particular bug in production.

Reply:
Added Buffer::startsWith and switch to using that.